src/util/sandbox.cpp

```diff
@@ -385,66 +401,76 @@ QString Sandbox::migrateOldSettings() {
+            kCFAllocatorDefault, QStringToCFString(legacySettingsPath), kCFURLPOSIXPathStyle, true);
     if (url) {
         CFErrorRef error = NULL;
-        CFDataRef bookmark = CFURLCreateBookmarkData(
+        if (sandboxContainerId != "") {
+            // Request permissions to old sandboxed settings path in sandbox and move the directory
+            CFDataRef bookmark = CFURLCreateBookmarkData(
                 kCFAllocatorDefault,
                 url,
                 kCFURLBookmarkCreationWithSecurityScope,
                 nil,
                 nil,
                 &error);
-        CFRelease(url);
-        if (bookmark) {
-            QFile oldSettings(oldPath);
-            if (oldSettings.rename(sandboxedPath)) {
-                qInfo() << "Sandbox::migrateOldSettings: Successfully "
-                           "migrated old settings from"
-                        << oldPath << "to new path" << sandboxedPath;
+            CFRelease(url);
+            if (!bookmark) {
+                QFile oldSettings(legacySettingsPath);
+                if (oldSettings.rename(sandboxedPath)) {
+                    qInfo() << "Sandbox::migrateOldSettings: Successfully "
+                            "migrated old settings from"
+                            << legacySettingsPath << "to new path" << sandboxedPath;
+                } else {
+                    qWarning() << "Sandbox::migrateOldSettings: Failed to migrate "
+                                "old settings from" << legacySettingsPath
+                                << "to new path" << sandboxedPath;
+                }
+                CFRelease(bookmark);
             } else {
-                qWarning() << "Sandbox::migrateOldSettings: Failed to migrate "
-                              "old settings from"
-                           << oldPath << "to new path" << sandboxedPath;
+                qWarning() << "Sandbox::migrateOldSettings: Failed to access old "
+                            "settings path" << legacySettingsPath
+                        << "Cannot migrate to new path" << sandboxedPath;
             }
         } else {
-            qWarning() << "Sandbox::migrateOldSettings: Failed to access old "
-                          "settings path"
-                       << oldPath << "Cannot migrate to new path"
-                       << sandboxedPath;
+            // Move old sandboxed settings path when not in sandbox to new path
+            QFile oldSettings(legacySettingsPath);
+            if (oldSettings.rename(sandboxedPath)) {
+                qInfo() << "Sandbox::migrateOldSettings: Successfully migrated old settings from"
+                    << legacySettingsPath << "to new path" << sandboxedPath;
+            } else {
+                qWarning() << "Sandbox::migrateOldSettings: Failed to migrate old settings from"
+                    << legacySettingsPath << "to new path" << sandboxedPath;
+            }
         }
-        CFRelease(bookmark);
     }
-#endif
```

Comment:
make sure the #endif is here before the return statement to avoid compiler warnings on other OSes